documentation/USAGE.rst

```diff
@@ -44,6 +44,50 @@ Python can be obtained from the :cite:`PythonReference`.
 As you gain experience, you may want to browse through the
 :ref:`FlagsAndEnvironmentVariables` that affect :term:`FiPy`.
 
+-------
+Logging
+-------
+
+Diagnostic information about a :term:`FiPy` run can be obtained using the
+:mod:`logging` module.  For example, at the beginning of your script, you
+can add::
+
+    >>> import logging
+    >>> log = logging.getLogger("fipy")
+    >>> console = logging.StreamHandler()
+    >>> console.setLevel(logging.INFO)
+    >>> log.addHandler(console)
+
+in order to see informational messages in the terminal.  To have more
+verbose debugging information save to a file::
+
+    >>> logfile = logging.FileHandler(filename="fipy.log")
+    >>> logfile.setLevel(logging.DEBUG)
+    >>> log.addHandler(logfile)
+
+    >>> log.setLevel(logging.DEBUG)
+
+To restrict logging to, e.g., information about the :term:`PETSc` solvers::
+
+    >>> petsc = logging.Filter('fipy.solvers.petsc')
+    >>> logfile.addFilter(petsc)
+
+More complex configurations can be specified by setting the
+:envvar:`FIPY_LOG_CONFIG` environment variable.  In this case, it is not
+necessary to add any logging instructions to your own script.  Example
+configuration files can be found in
+:file:`{FiPySource}/fipy/tools/logging/`.
+
+If `Solving in Parallel`_, the :mod:`mpilogging` package enables reporting
+which MPI rank each log entry comes from.  For example::
+
+    >>> from mpilogging import MPIScatterdFileHandler
```

Comment:
I installed mpilogging with `python -m pip install --user https://github.com/guyer/mpilogging/archive/51a816699e0f054ea8c2d75d567b42f8224112e2.tar.gz`

and got 

```
>>> from mpilogging import MPIScatterdFileHandler
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'MPIScatterdFileHandler' from 'mpilogging' (/home/wd15/git/fipy/.local/lib/python3.7/site-packages/mpilogging/__init__.py)
```

Reply:
There's a typo in USAGE.rst:
```python
>>> from mpilogging import MPIScatteredFileHandler
                                     ^
```

Reply:
Fixed